__tests__/app.test.js

```diff
@@ -151,4 +151,50 @@ describe('/api/articles', () => {
 				});
 		});
 	});
+});
+
+describe('/api/articles/:article_id/comments', () => {
+	describe('200: GET:', () => {
+		it('responds with status 200 and an array of comments with the right shape', () => {
+			return request(app)
+				.get('/api/articles/1/comments')
+				.expect(200)
+				.then(({ body: { comments } }) => {
+					expect(comments).toBeInstanceOf(Array);
+					expect(comments).toHaveLength(11);
+					comments.forEach(comment => {
+						expect(comment).toMatchObject({
+							comment_id: expect.any(Number),
+							body: expect.any(String),
+							article_id: 1,
+							author: expect.any(String),
+							votes: expect.any(Number),
+							created_at: expect.any(String),
+						});
+					});
+					expect(comments).toBeSortedBy('created_at', { descending: true });
+				});
+		});
+
+		it('responds with status 200 and an empty array when there is no comments', () => {
+			return request(app)
+				.get('/api/articles/2/comments')
+				.expect(200)
+				.then(({ body: { comments } }) => {
+					expect(comments).toEqual([]);
+				});
+		});
+	});
+
+	describe('404: GET:', () => {
+		it('responds with status 404 when given an article_id that does not exist yet', () => {
+			return request(app).get('/api/articles/99999/comments').expect(404);
```

Comment:
Shouldn't you be sending a message to the client too?

We need to make our errors useful in some way, an indication that the article doesn't exist or the article id was invalid (for the below) is massively helpful when designing an API

Reply:
it does, i just didn't want to rewrite the same test every time, just checking for status code seems sufficient if i have already tested for the error message in a previous test, this 404 is not any different to an earlier 404, but i added the test anyway as suggested since you guys do not have full context of what happens earlier